redbot/cogs/warnings/warnings.py

```diff
@@ -76,6 +83,39 @@ def __init__(self, bot: Red):
         else:
             await ctx.send(_("Custom reasons have been disabled."))
 
+    @warningset.command()
+    @commands.guild_only()
+    async def toggledm(self, ctx: commands.Context):
+        """Toggle warns being sent to the users DM's"""
+        guild = ctx.guild
+        toggle = await self.config.guild(guild).toggle_dm()
+        await self.config.guild(guild).toggle_dm.set(not toggle)
+        toggle = not toggle
+        if toggle:
+            await ctx.send("Warns will now be sent to users DM's")
+        else:
+            await ctx.send("Warns will not be sent to users DM's")
+
+    @warningset.command()
+    @commands.guild_only()
+    async def warnchannel(self, ctx: commands.Context, channel: discord.TextChannel):
```

Comment:
I think that it should be allowed to reset this setting (as in, set it to `None` when channel is not provided), so that bot would use `ctx.channel` in `[p]warn` command, when `togglechannel` is enabled and this is not set.